compiler/src/org.graalvm.compiler.core.amd64/src/org/graalvm/compiler/core/amd64/AMD64ArithmeticLIRGenerator.java

```diff
@@ -961,6 +964,18 @@ public Variable emitBitScanReverse(Value value) {
         return result;
     }
 
+    @Override
+    public Variable emitFusedMultiplyAdd(Value a, Value b, Value c) {
+        Variable result = getLIRGen().newVariable(LIRKind.combine(a, b, c).changeType(AMD64Kind.DWORD));
+        if (a.getPlatformKind() == AMD64Kind.QWORD) {
+            getLIRGen().append(new AMD64Ternary.ThreeOp(VFMADD231SD, AVXSize.QWORD, asAllocatable(a), asAllocatable(a), asAllocatable(b), asAllocatable(c)));
```

Comment:
Since we are issuing an LIG instruction, we can replace `AVXSize.QWORD` with `AVXSize.XMM`. Same for line 974. Also please replace the first `asAllocatable(a)` with `result`.